### DISCUSSION TITLE
Welcome to OpenVPN 3 Core Library Discussions!

### DISCUSSION THREAD

**Discussion by dsommers** [OpenVPN/openvpn3#298] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by salaros** [comment]
Why don't you release a plugin for native Windows 10/11 VPN (UWP / WinUI) client?

**Reply by lstipakov** [reply]
A few years ago we've played with UWP VPN API and made a proof-of-concept plugin, however we were not super happy with the performance - it was on par with our legacy kernel driver (tap-windows6). Besides, UWP as a technology appears to be dead as for now. If you need a performance you should use dco-win driver, which is now default in our community openvpn2 Windows client and probably in closed-source Connect client too.

**Comment by rizkyblackhat** [comment]
How can we build this library as a .lib or .dll instead of ovpncli.exe? We need the library and headers to be used in C++/VB .NET like a normal library. When we try to use the library directly by including the client and openvpn directories in the include path in Visual Studio IDE, we encounter a large number of errors.

**Reply by schwabe** [reply]
Currently this library is only useable as header library in C++. But it works fine with MSVC. You can try using it as swig wrapper with other languages though.
